Sources/Core/AWSClientRuntime/UserAgent/OSMetadata.swift

```diff
@@ -0,0 +1,43 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import ClientRuntime
+
+struct OSMetadata {
+    let family: PlatformOperatingSystem
+    let version: String?
+
+    init(family: PlatformOperatingSystem, version: String? = nil) {
+        self.family = family
+        self.version = version
+    }
+}
+
+extension OSMetadata: CustomStringConvertible {
+
+    var description: String {
+        guard let version = version, !version.isEmpty else {
+            return "os/\(family.userAgentName)"
+        }
+        return "os/\(family.userAgentName)#\(version.sanitizedForUserAgentToken)"
+    }
```

Comment:
Since "os/\(family.userAgentName)" is in both potential outputs can consider doing something like

```
var description: String {
    let versionSuffix = (version != nil && !version.isEmpty) ? "#\(version.sanitizedForUserAgentToken)" : ""
    return "os/\(family.userAgentName)\(versionSuffix)"
}
```

Reply:
I have restructured the construction of this string to be clearer.